OgreMain/include/OgreSceneLoaderManager.h

```diff
@@ -0,0 +1,65 @@
+#ifndef __SCENELOADERMANAGER_H__
+#define __SCENELOADERMANAGER_H__
+
+#include "OgreSingleton.h"
+#include "OgreSceneLoader.h"
+
+namespace Ogre {
+    
+    /** \addtogroup Core
+    *  @{
+    */
+    /** \addtogroup Resources
+    *  @{
+    */
+    /** This class defines an interface for registering and using SceneLoaders.
+    */
+    class _OgreExport SceneLoaderManager : public Singleton<SceneLoaderManager>
+    {
+    public:
+        SceneLoaderManager();
+        virtual ~SceneLoaderManager();
+        /** Register a new SceneLoader
+            @param name The name for the SceneLoader for lookup.
+            @param ext A StringVector containing the supported extensions for the
+                SceneLoader being registered.
+            @param sl Pointer to the SceneLoader instance.
+        */
+        void registerSceneLoader(const String& name, StringVectorPtr ext, SceneLoader *sl);
```

Comment:
please use `const StringVector&` instead of StringVectorPtr. See #668.

Otherwise the API looks good to me :+1: